src/main/java/by/onliner/page/ShoppingCartPage.java

```diff
@@ -61,4 +77,62 @@ public String getEmptyCartMassage() {
         logger.info(String.format("Displayed [%s] message", emptyCartMessage.getText()));
         return emptyCartMessage.getText();
     }
+
+    /***
+     * Get the item name from shopping cart list.
+     * @return name of the item
+     */
+    @Step("Read the item name")
+    public String getItemName() {
+        Waiter.waitForVisibility(itemName);
+        logger.info(String.format("Read the item name [%s] from shopping cart", itemName.getText()));
+        return itemName.getText().trim();
+    }
+
+    /***
+     * Check the complete order button fo visibility.
+     * @return visibility of button
+     */
+    @Step("Chek [Перейти к оформлению] button for visibility")
+    public boolean isCompleteOrderButtonVisible() {
+        boolean visibility = completeOrderButton.isDisplayed();
+        if (visibility) {
+            logger.info("'Перейти к оформлению' button is visible");
+        }
+        return visibility;
+    }
+
+    /***
+     * Add one more same item in the shopping cart by [+] button
+     */
+    @Step("Add one more same item in the shopping cart by [+] button")
+    public void clickQuantityInputPlusButton() {
+        Waiter.elementToBeClickable(quantityInputPlusButton);
+        quantityInputPlusButton.click();
+        logger.info("Click '+' button");
+        Waiter.sleep(2);
+    }
+
+    /***
+     * Get number from quantity input
+     * @return integer number from quantity input
+     */
+    @Step("Get number from quantity input")
+    public int getNumberFromQuantityInput() {
+        Waiter.waitForVisibility(quantityInput);
+        logger.info("Get number [" + quantityInput.getAttribute("value") + "] from quantity input");
+        return Integer.parseInt(quantityInput.getAttribute("value"));
+    }
+
+    /***
+     * Get price of item
+     * @return integer price of item
+     */
+    @Step("Get price")
+    public int getPrice() {
+        Waiter.waitForVisibility(itemPrice);
+        int price = Integer.parseInt(itemPrice.getText().substring(0, itemPrice.getText().length() - 6));
```

Comment:
Why are you using hardcoded indexes? Are you sure it won't change?
I would recommend to update that to smth more dynamic

Reply:
⭐

Reply:
Fidxed by using redex